examples/tests/valid/inflight_ref_resource_collection.w

```diff
@@ -1,16 +1,19 @@
-// NOTE: this test is supposed to fail once https://github.com/winglang/wing/pull/1610 is complete
-
 bring cloud;
 
 resource Test {
   array: Array<cloud.Bucket>;
+  bucket: cloud.Bucket;
 
   init() {
     this.array = [new cloud.Bucket()];
+    assert(this.array.length == 1);
+    this.array.at(0).add_object("foo.txt", "bar");
+    this.bucket = new cloud.Bucket() as "b2";
   }
 
   inflight test() {
-    assert(this.array.at(0).list().length == 0);
+    assert(this.array.at(0).list().length == 1);
+    assert(this.array.at(0).get("foo.txt") == "bar");
```

Comment:
I don't think we can compile this example at the moment, right?

It looks like we are just generating an annotation like this
```js
Test._annotateInflight("test", {"this.array": { ops: ["at"] }});
```
so the test will fail inflight 😞 

I reckon we need a bit more machinery on both sides. First, we need a way to represent information that says "here's a list of operations used on elements of this collection" in our inflight annotation data structure. For example:

```js
Test._annotateInflight("test", {"this.array": { elem_ops: ["list", "get"] }});
```

This says that we need "list" and "get" operation permissions from all of the buckets INSIDE the collection. From a bindings/permissions perspective, we don't care about what operations are called on the array itself (like `at`).

Second, the compiler needs to be able to obtain that information through analysis. I'm thinking that some operations that access elements of collections, like `map.get()` or `array.at()` need to be special cased in our compiler's binding checker. Note that we don't care about the actual index passed, or the key of the map, etc. -- `Map`, `Array`, etc. are all homogenous collections so we shouldn't treat the elements as pets / try to generate different permissions for them.

Anyways, these are just my initial thoughts of one direction I'd go in (probably beyond the scope of this PR) - curious if any parts of it resonate to you or not. :)

Reply:
We could also extend the approach I mention above to structs. For example, I'd like to write this in Wing:

```js
struct MyBuckets {
  east: cloud.Bucket;
  west: cloud.Bucket;
}
```

and based on usage, have specific annotations generated like:

```js
Test._annotateInflight("test", {"this.my_buckets": { key_ops: { east: ["list"], west: ["get"] } }});
```

But we can probably live without this for beta since a struct with resources isn't too different from a composite resource. (However, there is an important difference which is that a struct can be instantiated inflight while a resource cannot! ☝️)

Reply:
This is an error (for now).